runtime/runtime_test.go

```diff
@@ -241,6 +243,22 @@ func (i *testRuntimeInterface) SetProgram(location Location, program *interprete
 	return i.setProgram(location, program)
 }
 
+func (i *testRuntimeInterface) SetInterpreterSharedState(state *interpreter.SharedState) {
+	if i.setInterpreterSharedState == nil {
+		return
+	}
+
+	i.setInterpreterSharedState(state)
+}
+
+func (i *testRuntimeInterface) GetInterpreterSharedState() *interpreter.SharedState {
+	if i.getInterpreterShareState == nil {
+		return nil
+	}
```

Comment:
maybe panic here like the other testRuntimeInterface methods, since returning nil doesn't signify whether the operation failed or if the interface just doesn't have a fn pointer

Reply:
`GetInterpreterSharedState` is an optional function, the embedder does not need to implement it. The runtime still works if this function is not implemented